unittests/qis/QubitQISTester.cpp

```diff
@@ -424,4 +424,127 @@ CUDAQ_TEST(QubitQISTester, checkU3Adj) {
   }
 }
 
+// Test someone can build a library of custom operations
+CUDAQ_REGISTER_OPERATION(
+    /* Name */ CustomHadamard, /*NumTargets*/ 1, /*NumParameters*/ 0,
+    /* Unitary Generator */ {M_SQRT1_2, M_SQRT1_2, M_SQRT1_2, -M_SQRT1_2});
+CUDAQ_REGISTER_OPERATION(CustomX, 1, 0, {0, 1, 1, 0});
+CUDAQ_REGISTER_OPERATION(CustomCNOT, 2, 0,
+                         {1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 1, 0});
```

Comment:
This is the Big-endian representation of CNOT. Hence it applies as `CustomCNOT(target, control)` (not the CUDA-Q convention of `cnot(control, target)`.

Reply:
The latest update will ask the simulators about their bit ordering and convert if necessary